From eee2eba454fdb5d07a40f5df59d6af46847efe40 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 26 Aug 2024 23:48:26 -0700
Subject: [PATCH 1/3] llvm path update for rocm sdk builder

- check both the rocm/bin and rocm/llvm/bin
  directories

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe822b7..c878520 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,11 @@ message("-- ${BoldBlue}rocDecode Install Path -- ${CMAKE_INSTALL_PREFIX}${Colour

 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH} ${ROCM_PATH}/hip)
-set(CMAKE_CXX_COMPILER ${ROCM_PATH}/llvm/bin/clang++)
+if(EXISTS ${ROCM_PATH}/bin/clang++)
+  set(CMAKE_CXX_COMPILER ${ROCM_PATH}/bin/clang++)
+else()
+  set(CMAKE_CXX_COMPILER ${ROCM_PATH}/llvm/bin/clang++)
+endif()

 # rocDecode Default Options
 option(BUILD_WITH_AMD_ADVANCE "Build rocDecode for advanced AMD GPU Architecture" OFF)
--
2.41.1
